src/EditorFeatures/Core/Intents/IIntentsEditsSource.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Features.Intents;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Intents
+{
+    internal interface IIntentsEditsSource
+    {
+        /// <summary>
+        /// For an input intent, computes the edits required to apply that intent and returns them.
+        /// </summary>
+        /// <param name="intentRequestContext">the intents with the context in which the intent was found.</param>
+        /// <returns>the edits that should be applied to the current snapshot.</returns>
+        Task<IntentResult?> ComputeEditsAsync(IntentRequestContext intentRequestContext, CancellationToken cancellationToken);
```

Comment:
I believe this is essentially how we discussed the API, however I encountered a few issues when implementing.

1.  If intellicode wants to ask us for multiple intents, this won't support it (or multiple intents with different intent data, e.g. generate ctor with different fields).
2.  If we want to return multiple different kinds of actions for a single intent (for example, generate ctor can offer a ctor for all fields, or to delegate through an existing ctor).

Reply:
yes.  i think this should be a one->many request, or many->many request.  If we make it one->many, then they can call us multiple times.  i think that's likely fine, unelss there is a deep perf issue here, or we want to know all the set of inputs to pick the best set of outputs.

we'll have to discuss with them about this.